In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

df = df = pd.read_csv("titanic_limpio.csv")

In [2]:
porcentaje_supervivientes = df["survived"].mean() * 100

print(f"Supervivientes: {porcentaje_supervivientes:.2f}%")

Supervivientes: 41.29%


In [3]:
supervivencia_sexo = df.groupby("sex")["survived"].mean() * 100

print(supervivencia_sexo)

sex
female    73.972603
male      21.532091
Name: survived, dtype: float64


In [4]:
supervivencia_clase = df.groupby("pclass")["survived"].mean() * 100

print(supervivencia_clase)

pclass
1    63.333333
2    50.609756
3    25.935162
Name: survived, dtype: float64


In [5]:
edad_supervivencia = df.groupby("survived")["age"].mean()

print(edad_supervivencia)

survived
0    30.482418
1    28.299750
Name: age, dtype: float64


In [6]:
supervivencia_edad = df.groupby("franja_edad")["survived"].mean() * 100

print(supervivencia_edad)

franja_edad
Adulto    39.531250
Mayor     28.000000
Menor     54.545455
Name: survived, dtype: float64


In [7]:
sexo_clase = df.groupby(["sex", "pclass"])["survived"].mean() * 100

print(sexo_clase)

sex     pclass
female  1         96.774194
        2         91.666667
        3         47.244094
male    1         36.752137
        2         18.478261
        3         16.058394
Name: survived, dtype: float64


In [8]:
df.to_csv("titanic_limpio.csv", index=False)

1. ¿Qué porcentaje de pasajeros sobrevivió?

41,29%

2. ¿Cómo varía la supervivencia según el sexo?

La supervivencia en mujeres fue notablemente mas alta que la de los hombres

3. ¿Cómo varía la supervivencia según la clase?

La primera clase tiene una tasa de supervivencia del 63% aprox. La segunda clase del 50 y la tercera del 25 (aprox)

4. ¿Existe relación entre la edad y la probabilidad de supervivencia?

Parece ser que si, puesto que cuanto mas joven la persona mayor probabilidad de supervivencia (54% frente a 39% en adultos y 28 en mayores)

5. ¿Qué grupo de edad presenta una mayor proporción de supervivientes?

El grupo de menores de edad.

Conclusiones:
- La supervivencia no fue igual para todos los pasajeros y presenta diferencias según distintas características.
- El sexo muestra diferencias en las tasas de supervivencia.
- La clase del billete también presenta diferencias en la proporción de supervivientes.
- La edad permite observar diferencias entre los distintos grupos establecidos.
- El análisis conjunto de sexo y clase permite observar cómo ambas variables se relacionan con la supervivencia.